File: app/src/main/java/alireza/nezami/dependency_injection_without_dagger/java/approach3/data/remote/NetworkService.java
package alireza.nezami.dependency_injection_without_dagger.java.approach3.data.remote;

import android.content.Context;

/**
 * Dummy class to simulate the actual NetworkService using Retrofit or OkHttp etc
 */
public class NetworkService {

    private Context context;
    private String apiKey;

    public NetworkService(Context context, String apiKey) {
        // do the initialisation here
        this.context = context;
        this.apiKey = apiKey;
    }

    public String getDummyData() {
        return "NETWORK_DUMMY_DATA";
    }
}

File: app/src/main/java/alireza/nezami/dependency_injection_without_dagger/kotlin/approach3/data/remote/NetworkService.kt
package alireza.nezami.dependency_injection_without_dagger.kotlin.approach3.data.remote

import android.content.Context

/**
 * Dummy class to simulate the actual NetworkService using Retrofit or OkHttp etc
 */
class NetworkService
    (private val context: Context, private val apiKey: String) {
    val dummyData: String
        get() = "NETWORK_DUMMY_DATA"
}